patching/src/main/java/org/jboss/as/patching/management/LocalAgeoutHistoryHandler.java

```diff
@@ -68,20 +66,31 @@ public boolean accept(File dir, String name) {
     };
 
     @Override
-    public void execute(final OperationContext context, final ModelNode operation) throws OperationFailedException {
+    protected void execute(OperationContext context, ModelNode operation, InstallationManager instMgr, String patchStream) throws OperationFailedException {
+        try {
+            if (patchStream != null) {
+                final InstalledIdentity installedIdentity = instMgr.getInstalledIdentity(patchStream, null);
+                ageOutHistory(installedIdentity);
+            } else {
+                for (InstalledIdentity installedIdentity : instMgr.getInstalledIdentities()) {
+                    ageOutHistory(installedIdentity);
+                }
+            }
+        } catch (PatchingException e) {
+            throw new OperationFailedException(PatchLogger.ROOT_LOGGER.failedToLoadIdentity(), e);
```

Comment:
Is this failure due to a client mistake? If not, a RuntimeException of some sort should be thrown, as OFE represents a client mistake and is only logged in the server.log at DEBUG. Also, at some point we'll use exceptions to trigger proper HTTP response codes (versus current 500) in the http interface, and an OFE would result in a 4xx.

Don't worry about this for this patch; this doesn't look any different from other places in the patching code. My question is intended to trigger a general review at some point of throwing OFE in the patching resources.